### ISSUE TITLE
TCE: Launching the same kernel twice will ignore the global_work_offset on the second call 

### ISSUE THREAD

**Issue by nrother** [pocl/pocl#1135]
This is a problem in the TCE driver: When launching a kernel with `global_work_offset` set to all zero it will also be zero on all subsequent calls, even if a different value is specified in the other launches.

PyOpenCL Code to reproduce:

```py
#!/usr/bin/env python3

import numpy as np
import pyopencl as cl

device = cl.get_platforms()[0].get_devices()[0]
print(f"Using device {device}")

ctx = cl.Context(devices=[device])
queue = cl.CommandQueue(ctx)

with open("goffs.cl", 'r') as f:
    prg = cl.Program(ctx, f.read()).build()

out = np.empty((6,), dtype=np.int32)
out2 = np.empty((6,), dtype=np.int32)

mf = cl.mem_flags
out_cl = cl.Buffer(ctx, mf.WRITE_ONLY, out.nbytes)
out2_cl = cl.Buffer(ctx, mf.WRITE_ONLY, out2.nbytes)

knl = prg.test_goffs  # Use this Kernel object for repeated calls
knl(queue, (1,), None, out_cl, global_offset=(0,))
#knl(queue, (1,), None, out_cl, global_offset=(5,))
knl(queue, (1,), None, out2_cl, global_offset=(10,))

cl.enqueue_copy(queue, out, out_cl)
cl.enqueue_copy(queue, out2, out2_cl)

print(out)
print(out2)
```

**goffs.cl**
```cl
__kernel void test_goffs(__global int *out)
{
    out[0] = get_global_id(0);
    out[1] = get_global_id(1);
    out[2] = get_global_id(2);
    out[3] = get_global_offset(0);
    out[4] = get_global_offset(1);
    out[5] = get_global_offset(2);                
}
```

**Output**
```
[0 0 0 0 0 0]
[0 0 0 0 0 0]
```

**Expected output**
```
[0 0 0 0 0 0]
[10  0  0 10  0  0]
```

For other devices like pthreads this is working as expected. If the first call does not set the offset to zero it works as expected (see commented line).

**Comment by pjaaskel** [comment]
Can you try turning the kernel compiler cache off? It might be a problem with the cache key / directory name generation.

**Comment by nrother** [comment]
I just tried it with `POCL_KERNEL_CACHE=0` and it is still happening. Or is there another way to disable the cache?

**Comment by TopiLeppanen** [comment]
Yeah, it's related to the driver not bothering to reload the program if it thinks the kernel is the same. It didn't take global offset into account, so I added that.

The reason this worked in AlmaIF is that I had disable the similar check earlier for diffrerent reason, which was related to the kernel being freed and then different kernel created that happened to have the same pointer.

**Comment by nrother** [comment]
Fixed by #1139
